docs/decisions/0009-course-catalog-info-changed-design.rst

```diff
@@ -0,0 +1,37 @@
+8. Event design of the COURSE_CATALOG_INFO_CHANGED signal
+=========================================================
+
+Status
+------
+
+Provisional
+
+Context
+-------
+
+The COURSE_CATALOG_INFO_CHANGED signal is used by Studio to convey that information relevant to the course catalog has changed. It was created with an eye towards eventually replacing the refresh-course-metadata batch job, which syncs all data between edx-platform and course-discovery.
+
+Refresh-course-metadata functions differently depending on whether or not the system has Publisher enabled. Specifically, after requesting information from the /courses endpoint in Studio, Discovery will ignore certain fields if Publisher is enabled. These fields are mostly part of the `media` attribute of the API response. They are sent in a variety of ways (absolute urls, paths, sometimes divided by size, etc.)
+
+
+Decision
+--------
+
+The COURSE_CATALOG_INFO_CHANGED will only contain the information necessary to work in a Publisher-enabled environment. In particular, this means it will not contain the `media` field usually present in the Studio /courses API endpoint.
+
+In Discovery, if Publisher is not enabled, the consumer will simply ignore the event and not try to update anything.
```

Comment:
I may be missing something, but I don't think we should do this. It feels like we should error instead (for now). It feels like the biggest problem would be a misconfiguration where you are in a publisher environment, but these events were being ignored. Maybe we'd just want to improve the error messaging? Are there cases where it is important for this consumer to be running and not doing these updates?

Reply:
We would definitely want to at least improve the error messaging so we don't flood the logs with a bunch of key errors. I think it's more worth a warning than an error, which would basically be ignore-but-log-better.

Reply:
That sounds fine. I just didn't want to silently ignore a likely problematic state.

Reply:
fixed